test/puppetlabs/puppetdb/scf/storage_test.clj

```diff
@@ -1204,41 +1212,44 @@
           (is (= false (maybe-activate-node! certname (now))))
           (is (= (query-certnames) [{:certname certname :deactivated nil}])))))))
 
-(deftest-db node-staleness-age
-  (testing "retrieving stale nodes based on age"
-    (let [query-certnames #(query-to-vec ["select name, deactivated from certnames order by name"])
-          deactivated?    #(instance? java.sql.Timestamp (:deactivated %))]
-
-      (testing "should return nothing if all nodes are more recent than max age"
-        (let [catalog (:empty catalogs)
-              certname (:certname catalog)]
-          (add-certname! certname)
-          (replace-catalog! (assoc catalog :producer_timestamp (now)) (now))
-          (is (= (stale-nodes (-> 1 days ago)) [])))))))
-
-(deftest-db node-stale-catalogs-facts
-  (testing "should return nodes with a mixture of stale catalogs and facts (or neither)"
-    (let [mutators [#(replace-catalog! (assoc (:empty catalogs) :certname "node1") (-> 2 days ago))
-                    #(replace-facts! {:certname "node1"
-                                      :values {"foo" "bar"}
-                                      :environment "DEV"
-                                      :producer_timestamp "2014-07-10T22:33:54.781Z"
-                                      :timestamp (-> 2 days ago)})]]
-      (add-certname! "node1")
-      (doseq [func-set (subsets mutators)]
-        (dorun (map #(%) func-set))
-        (is (= (stale-nodes (-> 1 days ago)) ["node1"]))))))
-
-(deftest-db node-stale-reports
-  (testing "should not return a node with a recent report and nothing else"
+(deftest-db fresh-node-not-expired
+  (testing "fresh nodes are not expired"
+    (let [catalog (:empty catalogs)
+          certname (:certname catalog)]
+      (add-certname! certname)
+      (replace-catalog! (assoc catalog :producer_timestamp (now)) (now))
+      (is (= [] (expire-stale-nodes (-> 3 days .toPeriod))))
```

Comment:
Actually, I had that first, but then remembered that if you just check empty, you can't see what was wrong when it fails.